sdk/src/index.ts

```diff
@@ -1,177 +1,164 @@
-  const psp22 = await PSP22.load(api, network, token0Address)
-  const feeTier = newFeeTier(6000000000n, 10n)
+  const createPoolResult = await invariant.createPool(account, poolKey, initSqrtPrice)
+  console.log(createPoolResult.hash) // print transaction hash
 
-  const poolKey = await newPoolKey(token0Address, token1Address, feeTier)
+  // token y has 12 decimals and we want to add 8 actual tokens to our position
+  const tokenYAmount = 8n * 10n ** 12n
 
-  await invariant.addFeeTier(account, feeTier)
+  // set lower and upper tick, we want to create position in range [-10, 10]
+  const lowerTick = -10n
+  const upperTick = 10n
 
-  await invariant.createPool(account, poolKey, 1000000000000000000000000n, 0n)
+  // calculate amount of token x we need to give to open position
+  const { amount: tokenXAmount, l: positionLiquidity } = getLiquidityByY(
+    tokenYAmount,
+    lowerTick,
+    upperTick,
+    initSqrtPrice,
+    true
+  )
 
-  await psp22.setContractAddress(token0Address)
-  await psp22.approve(account, invariant.contract.address.toString(), 10000000000n)
-  await psp22.setContractAddress(token1Address)
-  await psp22.approve(account, invariant.contract.address.toString(), 10000000000n)
+  // print amount of token x and y we need to give to open position based on parameteres we passed
+  console.log(tokenXAmount, tokenYAmount)
 
-  const pool = await invariant.getPool(account, token0Address, token1Address, feeTier)
+  // approve transfers of both tokens
+  await psp22.setContractAddress(poolKey.tokenX)
+  await psp22.approve(account, invariant.contract.address.toString(), tokenXAmount)
+  await psp22.setContractAddress(poolKey.tokenY)
+  await psp22.approve(account, invariant.contract.address.toString(), tokenYAmount)
 
-  await invariant.createPosition(
+  // open up position
+  const createPositionResult = await invariant.createPosition(
     account,
     poolKey,
-    -10n,
-    10n,
-    1000000000000n,
-    pool.sqrtPrice,
-    pool.sqrtPrice
+    lowerTick,
+    upperTick,
+    positionLiquidity,
+    initSqrtPrice,
+    0n
   )
+  console.log(createPositionResult.hash) // print transaction hash
 
-  // deploy wrapped azero
-  const wazero = await WrappedAZERO.deploy(api, network, account)
+  // here we want to swap 6 token0
+  // token0 has 12 decimals so we need to multiply it by 10^12
+  const amount = 6n * 10n ** 12n
 
-  await transferBalance(api, account, testAccount.address, 1000000000000)
-  console.log('account balance: ', (await getBalance(api, account.address)).balanceFormatted)
-  console.log(
-    'test account balance: ',
-    (await getBalance(api, testAccount.address)).balanceFormatted
-  )
+  // approve token x transfer
+  await psp22.setContractAddress(poolKey.tokenX)
+  await psp22.approve(account, invariant.contract.address.toString(), amount)
+
+  // get estimated result of swap
+  const quoteResult = await invariant.quote(account, poolKey, true, amount, true, getMinSqrtPrice())
```

Comment:
IMO, we can simplify the `invariant,quote` interface by removing `target_sqrt_price`. This parameter could be automatically set as either `min_sqrt_price` or `max_sqrt_price` depending on the swap direction (`xToY`).